pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -431,7 +433,14 @@ func mountConfigMapValuesFiles(templateSpec *corev1.PodSpec, c *corev1.Container
 		fileName := reconcilermanager.HelmConfigMapRef
 		mountPath := filepath.Join("/etc/config", vf.Name, vf.ValuesFile)
 		valuesFiles = append(valuesFiles, filepath.Join(mountPath, fileName))
-		volumeName := "configmap-vol-" + strings.ToLower(vf.Name) + fmt.Sprintf("-%d", i)
+		volumeName := fmt.Sprintf("vol-%d-", i) + strings.ToLower(vf.Name)
+
+		if len(volumeName) > volumeNameLengthLimit {
+			// trim the volume name as we only have a length limit of 63, even though
+			// the ConfigMap name can be much longer.  The prefix
+			// containing the index i should be sufficient to uniquify the value.
+			volumeName = volumeName[:volumeNameLengthLimit]
```

Comment:
What happens if the name collide after trimming?

Reply:
The volume name is prefixed by fmt.Sprintf(“vol-%d-“, i), where i is the index of the volumeFileSource that we are creating this volume from, so there shouldn’t be any name collisions.

Reply:
I added a test as well so that we can be confident

Reply:
Do users care about volume names? Should we just name it `vol-<INDEX>`?

Reply:
SGTM. I updated it to `valuesFile-vol-<INDEX>` to be slightly more descriptive